src/components/metadata/metadata-code.js

```diff
@@ -0,0 +1,48 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import hljs from 'highlight.js/lib/core';
+import python from 'highlight.js/lib/languages/python';
+import yaml from 'highlight.js/lib/languages/yaml';
+import modifiers from '../../utils/modifiers';
+import './styles/metadata-code.css';
+
+hljs.registerLanguage('python', python);
+hljs.registerLanguage('yaml', yaml);
+
+/**
+ * A highlighted code panel
+ */
+export const MetaDataCode = ({
+  visible = true,
+  value = '',
+  sidebarVisible
+}) => {
+  const codeRef = useRef();
+  const [highlighted, setHighlighted] = useState();
+
+  useEffect(() => {
+    const detected = hljs.highlightAuto(value);
+    const language = detected.language || detected.second_best.language;
+    setHighlighted(hljs.highlight(language, value).value);
+  }, [codeRef, value]);
```

Comment:
Ah good catch I can remove the effect after since I switched it away from hljs rendering itself.

So it could just be synchronous, but actually I've swapped it out to `useMemo` instead for performance.